src/LLVM_Output.cpp

```diff
@@ -156,6 +156,10 @@ void emit_file(llvm::Module *module, const std::string &filename, llvm::TargetMa
     llvm::TargetMachine *target_machine = get_target_machine(module);
     internal_assert(target_machine) << "Could not allocate target machine!\n";
 
+    #if LLVM_VERSION >= 37
+    std::unique_ptr<llvm::raw_fd_ostream> out(new_raw_fd_ostream(filename));
```

Comment:
It makes sense to move it, but I also agree with @zvookin, it might be a good idea to just move the #Ifdef around the entire function.